src/client/dfuse/dfuse_core.c

```diff
@@ -270,18 +270,6 @@ _ph_free(struct dfuse_pool *dfp)
 
 	if (daos_handle_is_valid(dfp->dfp_poh)) {
 		rc = daos_pool_disconnect(dfp->dfp_poh, NULL);
-		/* Hook for fault injection testing, if the disconnect fails with out of memory
-		 * then simply try it again, only what might have happened is that the first
-		 * call might have disconnected, but then failed to notify about the disconnect,
-		 * in which case the subsequent call will return -DER_NO_HDL, if that's the case
-		 * then this is expected, if odd, behavior so silence that case and just return
-		 * success.
-		 */
-		if (rc == -DER_NOMEM) {
-			rc = daos_pool_disconnect(dfp->dfp_poh, NULL);
```

Comment:
I think you need this bit but not the next two lines.

Reply:
Actually I'm not sure myself now as this code didn't appear to actually call pool_disconnect a second time, just adjust the rc value.  Lets see what testing returns.

Reply:
Looks like fault injection passed as-is.